OpenTelemetry (#9709)
fishtown-analytics/docs.getdbt.com: website/docs/reference/events-logging.md

## Structured logging

<VersionBlock firstVersion="2.0">

> The <Constant name="fusion_engine" /> emits structured run telemetry using [OpenTelemetry](https://opentelemetry.io/) conventions instead of <Constant name="core" />'s JSON event logs. Use `--log-format otel` or the options in [<Constant name="fusion" /> telemetry and observability](/reference/telemetry-observability) to consume it locally. On the <Constant name="dbt_platform" />, you can [download OTel logs](/docs/deploy/run-visibility#access-logs) from Fusion job runs.

Reviewer: ```suggestion
The <Constant name="fusion_engine" /> emits structured run telemetry using [OpenTelemetry](https://opentelemetry.io/) conventions instead of <Constant name="core" />'s JSON event logs. To view it locally, use `--log-format otel` or the options in [<Constant name="fusion" /> telemetry and observability](/reference/telemetry-observability#available-output-formats).

In <Constant name="dbt_platform" />, you can [download OTel logs](/docs/deploy/run-visibility#access-logs) from <Constant name="fusion" /> job runs.
```

}
```

</VersionBlock>

## Python interface

<VersionBlock lastVersion="1.99">

Older versions of `dbt-core` made available a full history of events fired during an invocation, in the form of an `EVENT_HISTORY` object.

When [invoking dbt programmatically](programmatic-invocations#registering-callbacks), it is possible to register a callback on dbt's `EventManager`. This allows access to structured events as Python objects, to enable custom logging and integration with other systems.

The Python interface into events is significantly less mature than the structured logging interface. For all standard use cases, we recommend parsing JSON-formatted logs.

> </VersionBlock>

Reviewer 2: If you're going to put the Python interface header outside the version block, there needs to be some content for v2 readers. Could be a link to https://github.com/dbt-labs/dbt-core/issues/13102

}
```

</VersionBlock>

## Python interface

<VersionBlock firstVersion="2.0">

> <Constant name="fusion_engine" /> does not use the same Python event interface as <Constant name="core" />. A Python API with Rust bindings for programmatic invocations is [in development](https://github.com/dbt-labs/dbt-core/issues/13102).

Reviewer: sugg active voice
```suggestion
The <Constant name="fusion_engine" /> doesn't share <Constant name="core" />'s Python event interface. 

We are currently [developing](https://github.com/dbt-labs/dbt-core/issues/13102) a Python API with Rust bindings for programmatic invocations.
```
